WordPress/gutenberg
# Truncate: Convert component to TypeScript

## What?
Converts the `Truncate` component to TypeScript.

## Why?
Part of the @wordpress/components's TypeScript migration (https://github.com/WordPress/gutenberg/issues/35744).

## How?
Refactors the current `Truncate` to TypeScript.

## Testing Instructions
1. Run npm run dev and ensure no typing errors
2. Fire up the Storybook and confirm the `Truncate` continues to function as expected
